src/components/application_manager/src/message_helper/message_helper.cc

```diff
@@ -1789,12 +1789,16 @@ void MessageHelper::SendOnAppUnregNotificationToHMI(
 
 smart_objects::SmartObjectSPtr MessageHelper::GetBCActivateAppRequestToHMI(
     ApplicationConstSharedPtr app,
-    const protocol_handler::SessionObserver& session_observer,
     const policy::PolicyHandlerInterface& policy_handler,
     hmi_apis::Common_HMILevel::eType level,
     bool send_policy_priority,
     ApplicationManager& app_mngr) {
   DCHECK_OR_RETURN(app, smart_objects::SmartObjectSPtr());
+  if (hmi_apis::Common_HMILevel::NONE == level) {
```

Comment:
Isnt the app in NONE before it is activated? Can you please explain this if condition and early return?

Reply:
This is the requested level, not the current level. This condition is to prevent `BC.ActivateApp(NONE)`, since this would be used to deactivate the application instead. `BC.CloseApplication` is used instead.